wordpress-mobile/WordPress-Android
# Switch between editors while editing the post

Fixes #9037 

This PR adds the ability to switch between Aztec and Gutenberg while editing the post/page.

Implementation details:

* The `EditPostActivity` is let to finish and then we restart it with some proper flags
* Switching this way means that the post will at least get saved locally and then reloaded (in the "other" editor). It also means that Tracks that are triggered in the normal end/start of the editing Activity will get triggered here too. Hopefully the noise added will not be too harmful for the Tracks collected.
* The appbar 3-dots menu now has a "Switch to ... editor" option. Depending on the current editor and the content, the appropriate menu option is shown. See https://github.com/wordpress-mobile/WordPress-Android/issues/9037#issuecomment-463661960 for the conditions.

## To test A:
1. Open a GB post
2. Open the 3-dots menu and verify that the "Switch to Classic Editor" is there. Tap it.
3. Verify that the post is now opened in Aztec (look for the "H" button in the format bar)
4. Open the 3-dots menu and verify that the "Switch to Block Editor" is now there. Tap it.
5. Verify that the post is now opened in Gutenberg again (look for the undo/redo buttons in the format bar)
6. Make some change to the post (title or content)
7. Switch to Classic via the 3-dot menu
8. Verify that the changes made it to the Aztec
9. Make some more changes
10. Change to Gutenberg and verify that the new changes are there

## To test B:

1. Enable "Use Block Editor for new posts" in the App Settings
2. Create a new post. Verify it's Gutenberg.
3. Switch to Aztec via the 3-dot menu and verify that Aztec loads
4. Switch to Gutenberg via the 3-dot menu and verify that Gutenberg loads

## To test C:

1. *Disable* "Use Block Editor for new posts" in the App Settings
2. Create a new post. Verify it's Aztec.
3. Switch to Gutenberg via the 3-dot menu and verify that Gutenberg loads
4. Switch to Aztec via the 3-dot menu and verify that Aztec loads

## To test D:

Same as A but for pages.

## Update release notes:
